clients/web/src/components/calls/detail/call-detail-view.tsx

```diff
@@ -0,0 +1,49 @@
+import { For, JSXElement } from "solid-js";
+import { Span } from "~/lib/connection/monitor";
+import { CallDetailTrace } from "./call-detail-trace";
+import { CallDetailArgs } from "./call-detail-args";
+
+type Props = {
+  name: string;
+  spanChildren: Span[];
+  valuesSectionTitle: string;
+  values: (string | object)[];
+  children: JSXElement;
+};
+
+export function CallDetailView(props: Props) {
+  const closedSpans = () => props.spanChildren.filter((s) => s.closedAt > 0);
+
+  const durations = () => {
+    return {
+      start: Math.min(...closedSpans().map((s) => s.createdAt)),
```

Comment:
I totally agree with your point here. I used your suggestion and created a function from it, to not clutter/ blow up the component (prettier spreads out the solution you suggested quite a bit). 

Some background:
The detail view is the panel on the right side. So it will not be called too rapidly, but only on user action. The map functions are still a relic of the prototyping phase and since the amount of times the detail view will be called is fairly limited I refrained from optimizing.